### ISSUE TITLE
Name match not seeming to work when scraping c64 games with screenscraper module

### ISSUE THREAD

**Issue by RetroD** [muldjord/skyscraper#112]
Hello,
I have been trying to scrape my c64 games collection using the screenscraper module and found that many games were missed.
I am using  version 3.0.1 on retropie (stretch, installed a few days ago and with up to date startup script).
I have installed Skyscraper from the 'Retropie Setup' interface.

From what I understood by reading docs and reading various online posts, the screenscraper module will compare the checksums and the exact file names, and should be successful if EITHER of these match.

To try and figure out what's going on I have been experimenting the file named 'Burger Time (1984)(Interceptor Micros).zip' and also its unzipped version 'Burger Time (1984)(Interceptor Micros).prg'.
The unzipped version was a successful match, but the zip was not. I also tried renaming the zip to 'Burger Time.zip' but still no joy. (I think the string within brackets gets ignored anyway, so it wouldn't make a difference?)

The game name on the screenscraper website is listed as 'Burger Time', and I also downloaded the games.cvs from the site and its the same name.

Shouldn't the file 'Burger Time (1984)(Interceptor Micros).zip' have been a match in this case? I assume that the .prg worked because of a checksum match, but it would appear that the name match is not working?

This is the command I run (for some reason I cannot simply type 'Skyscraper' - it says it doesn't recognise the command, but that's a different issue):
pi@retropie:~ $ /opt/retropie/supplementary/skyscraper/Skyscraper -p c64 -s screenscraper 'Burger Time (1984)(Interceptor Micros).zip'

The output I got was:
#1/1 (T1) Pass 1 Pass 2 Pass 3 Pass 4 Pass 5 ---- Game 'Burger Time (1984)(Interceptor Micros)' not found :( ----

**Comment by muldjord** [comment]
If it doesn't match anything it's because it isn't in their database, it's as simple as that. File name matches only work if the file only exists once in their database (and it *has* to exist, this is not a search, it a precise match, that's how screenscraper works), so if another platform has the same filename it has to match with a checksum instead. If that is not the case, you won't get a hit. It works by unzipping it because the .prg file itself matches a checksum. You can use the `--unpack` option to force it to unpack files before checksumming them. But the zip must *only* contains exactly one file. Otherwise that won't work.

So basically this is an issue you have to report to the screenscraper database.

**Comment by RetroD** [comment]
@muldjord Thank you for the reply, I will try the - - unpack option, sounds useful.

So, just to make sure I understand, the filename match will not work if say, they have the same game name for the c64 system AND the amiga, and will only rely on the checksum in this case? I thought that when I scrape on a specific system it would look for that filename for that specific system on their database? (but maybe you can't query that) 

If that's not the case, could this be a feature that could be later implemented?

As for the file being in their database, it does exist. I can see it on their Web page and can get the entry when downloading their  .csv file with the games list for that system. 

@bleuge thank you i will use that option to see if I discover anything new

**Comment by muldjord** [comment]
> So, just to make sure I understand, the filename match will not work if say, they have the same game name for the c64 system AND the amiga, and will only rely on the checksum in this case? 

Yes.

> If that's not the case, could this be a feature that could be later implemented?

Never say never, but there are several issues with this that I won't detail here. So current answer is "probably not". :)

If a match isn't returned it's one of the two reasons I listed above. If you find a file with a checksum that matches one of their files *exactly* and it doesn't return a match, please let me know all the details of that file. Let me be clear, this is about the file itself, not if they are zipped. PRG or other files that are zipped where the checksum of the zip file itself is not in their database will only provide a result with the `--unpack` option enabled.
